### PR TITLE
Rewrite installation instructions

### LINKED ISSUE #3864: Improvement on the python-dev version
#### Preliminaries

- [x] I have followed the latest version of the
      [installation instructions](https://docs.manim.community/en/stable/installation.html).
- [x] I have checked the [installation FAQ](https://docs.manim.community/en/stable/faq/installation.html) and my problem is either not mentioned there,
      or the solution given there does not help.

## Description of error

Hi!
I followed the installation instructions but failed to pip install manim.
Some header files where missing even though I installed `python3-dev`.

It turns out, that it might be necessary to specifically install it for your python version.

This probably occurred, because I have several python versions installed.

So for me, installing `python3.12-dev` did the trick.

It might be good to mention this in the docs...

## Installation logs

      /tmp/pip-install-cceed8ly/manimpango_6dbecf73f2be40eab291ac24b3aeea55/manimpango/cmanimpango.c:83:10: fatal error: Python.h: No such file or directory
         83 | #include "Python.h"
            |          ^~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]

<details><summary>Terminal output</summary>

```
PASTE HERE OR PROVIDE LINK TO https://pastebin.com/ OR SIMILAR
```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (with version, e.g., Windows 10 v2004 or macOS 10.15 (Catalina)): Ubuntu 22.04
- RAM:
- Python version (`python/py/python3 --version`): 3.12
- Installed modules (provide output from `pip list`):
```
PASTE HERE
```
</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020):
+ Installed LaTeX packages:

</details>

### PR DESCRIPTION
**Pull request by JasonGrace2282**
#3501 removed the ffmpeg dependency - this meant that our installation docs could be drastically simplified. This an attempt to do that. Specifically
- Merges macOS and Windows installation instructions into one
- ~~Leaves Optional Dependency stuff in their own file depending on the system.~~ **EDIT: They're now in tabs - see behackl's comment**

Further improvement ideas welcome!

Also closes #3864

https://manimce--3930.org.readthedocs.build/en/3930/installation/local.html

### CODE REVIEW

**Comment by behackl** [comment]
I wonder whether we should collapse the installation instructions for Windows / MacOS / Linux back onto one single page and make use of code tabs https://sphinx-design.readthedocs.io/en/furo-theme/tabs.html#tabbed-code-examples to switch out the respective instructions?

With all the externally managed Python environments around it might also be worth considering just recommending an installation via uv; `uv tool install manim`. This would, however, probably increase friction as users would then look up the path to the tool env and feed that into their IDEs ... 🤔

**Comment by JasonGrace2282** [comment]
> I wonder whether we should collapse the installation instructions for Windows / MacOS / Linux back onto one single page and make use of code tabs https://sphinx-design.readthedocs.io/en/furo-theme/tabs.html#tabbed-code-examples to switch out the respective instructions?

Just took a look and it looks amazing! Definitely agree with using tabs for different operating systems.



> With all the externally managed Python environments around it might also be worth considering just recommending an installation via uv; `uv tool install manim`. This would, however, probably increase friction as users would then look up the path to the tool env and feed that into their IDEs ... 🤔

The idea of just `uvx manim` and calling it a day does seem pretty cool, but I agree users might have trouble linking the uv tool environment to their IDE. I think a better approach might be to just tell users to do something like
```
uv init project-name
cd project-name
uv add manim
uv run manim -p ...
```
And then most IDE's will recognize that a `.venv` folder exists in the current directory. Not as clean as a simple `uv tool install manim`, but it avoids that pain point.

**Comment by JasonGrace2282** [comment]
I just pushed a commit changing the recommendation to `uv`, would be glad to hear thoughts.

**Comment by behackl** [comment]
I very much like what you have done here, thank you for your efforts!

After some more thinking, there are some details I would change (I can push a commit -- or a separate branch for a 1:1 comparison):

- I like `uv` being used, a lot -- in particular as installing it is very straightforward on all of our platforms.
- I think the best practice that people should follow when creating manim projects is creating a local virtual environment to which they add manim as a dependency. Basically: `uv init my-manim-projects && cd my-manim-projects && uv add manim`. As long as we do not officially support 3.13, we will need to recommend pinning the python version to, say, 3.12. Installing manim as a global tool (or generally, installing it without using uv) can be recommended further down on the page.
- I'd shorten the optional dependencies section even further, just recommand MikTeX for Windows, MacTeX for MacOS, and texlive via the system's package manager for linux.
- The section on Conda should be rewritten to include a mention to homebrew too; maybe "Alternate System Package Managers" or so.

But all in all, I definitely agree that this sort of unified set of instructions is the way to go. We should look into setting up redirects from the old /windows, /linux, /macos pages to the new /local (or /pip?) page, otherwise we will break a lot of static guides.

**Comment by JasonGrace2282** [comment]
Feel free to push a commit, I probably won't be able to work on this PR until next weekend.
I also agree with setting up redirects, but I'm not quite sure how to do it (I don't think we can change the nginx configuration for readthedocs?)

**Comment by behackl** [comment]
I've pushed a somewhat orthogonal suggestion to this branch, the corresponding site is available at https://manimce--3930.org.readthedocs.build/en/3930/installation/pip.html.

I have not verified that all of the linux instructions really work exactly like they are stated here, but all in all I am pretty happy with where this is moving. I largely drew from @JasonGrace2282's first draft and tried to trim a few (in my opinion) superfluous branches while trying to keep the standard case somewhat succinct.

**Review by behackl — APPROVED** [review]
Tested the redirects, everything seems to be working.

File: docs/source/installation/uv.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+```
+uv init --python 3.12 manimations
+cd manimations
+uv add manim
+```
+To change the version for an existing package, you will need to
+edit two files contained in the root folder of your project:
+in `.python-version`, change the version number to the one you want
+to use (e.g., `3.12`). And secondly, in `pyproject.toml`, the
+`requires-python` entry needs to be updated such that your chosen
```
Comment (JasonGrace2282) [inline docs/source/installation/uv.md:281]: I'm not sure about the `requires-python` entry, but `uv python pin` should take care of most of the stuff.

File: docs/source/tutorials/quickstart.rst [hunk 2]
```diff
@@ -3,13 +3,19 @@ Quickstart
...
 .. note::
- Before proceeding, install Manim and make sure it's running properly by
+ Before proceeding, install Manim and make sure it is running properly by
  following the steps in :doc:`../installation`. For
  information on using Manim with Jupyterlab or Jupyter notebook, go to the
  documentation for the
  :meth:`IPython magic command <manim.utils.ipython_magic.ManimMagic.manim>`,
  ``%%manim``.
 
+ Also, very important: if you installed Manim in the recommended way, using the
```
Comment (JasonGrace2282) [inline docs/source/tutorials/quickstart.rst:13]: Maybe put this in a `.. important::` block?

**Review by JasonGrace2282 — COMMENTED** [review]
Thanks, Benjamin, for pushing the PR forward!
I left one or two comments from a quick readthrough, but overall, I'm happy to merge it in its current state.

File: docs/source/installation/uv.md (hunk 1 above)
Comment (JasonGrace2282) [reply docs/source/installation/uv.md:281]: Did some testing, `uv python pin` errors if it doesn't satisfy the `requires-python`.

**Pull request merged**